tests/unit/string_lib_test.c

```diff
@@ -496,6 +497,78 @@ static void test_string_to_long_compatibility(void)
     // Old function (StringToLongUnsafe) does not allow trailing whitespace
 }
 
+#define ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG(str, expected) \
+    assert_int_equal(StringDecimalToLong(str, &value_out), 0); \
+    assert_int_equal(value_out, expected)
+
+static void test_string_decimal_to_long(void)
+{
+    long value_out;
+
+    // Basic usage:
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("0.0", 0);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("-0.0", 0);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("+0.0", 0);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("123.0", 123);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("+123.0", 123);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("-123.0", -123);
+
+    // Leading spaces:
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG(" 0.0", 0);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG(" 123.0", 123);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG(" -123.0", -123);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("           0.0", 0);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("           123.0", 123);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("           -123.0", -123);
+
+    // Trailing spaces:
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("9.0 ", 9);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("789.0 ", 789);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("-789.0 ", -789);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("9.0           ", 9);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("789.0         ", 789);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("-789.0        ", -789);
+
+    // More spaces:
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("   0.0     ", 0);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("   -0.0    ", 0);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("   123.0  ", 123);
+
+    // Space separated numbers:
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("   123.456 78   9  ", 123);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("123.0 45", 123);
+}
+
+#define ASSERT_INT_NOT_EQUAL_STRING_DECIMAL_TO_LONG(str, expected) \
+    assert_int_not_equal(StringDecimalToLong(str, &value_out), 0);
```

Comment:
You never use this macro. Also, the name is not good and the `expected` parameter is not used.

```suggestion
#define ASSERT_STRING_DECIMAL_TO_LONG_ERRORS(str)                \
{                                                                \
    long *bad_addr = (long *) 1; /* Should not be written to */  \
    assert_int_not_equal(StringDecimalToLong(str, bad_addr), 0); \
}
```